### PR TITLE
RATEST-275:NEW E2E Test: Manage OCL Subscription Workflow

### PR DESCRIPTION
**Pull request by jwnasambu**
Description of what I did:

I created the following files:

-  open-Concept-Lab.yml file
- made changes on package.json file
- modified ConfigureMetadataPage
- Created OpenConceptLabPage
- Created OpenConceptLabSuccessPage
- Created SubscriptionPage
- Modified RunTest
- Created OpenConceptLabSteps
- Created openConceptLab.feature

Issue I worked on:

https://issues.openmrs.org/browse/RATEST-275

### CODE REVIEW

**Comment by jwnasambu** [comment]
@kdaud , @sherrif10 kindly this is the final work. Feel free to review at your convinient time please!

**Review by kdaud — APPROVED** [review]
cc: @sherrif10

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/OpenConceptLabSteps.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     } 
+    
+    @And("a user edits the URL of a released dictionary")
+    public void editDictionaryUrl() {
+        subscriptionPage.enterSubscriptionURL("https://api.qa-refapp.openmrs.org/openmrs/dictionary/concept.htm?conceptId=162172");
+    }   
+
+    @And("a user edits the Token url")
+    public void editTokenUrl() {
+        subscriptionPage.enterTokenURL("bd022mad6d3df24f5c42ewewa94b53a23edf6fff8r");
```
Comment (sherrif10) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/OpenConceptLabSteps.java:120]:
```suggestion
        subscriptionPage.enterTokenURL(EDITTED_TOKEN_URL);
```

**Review by sherrif10 — COMMENTED** [review]
This is greatly improved @jwnasambu  . Well done for the milestone. However one simply slight improvement. 
can we also add private static final String EDITTED_TOKEN_URL = "bd022mad6d3df24f5c42ewewa94b53a23edf6fff8r";

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/OpenConceptLabSteps.java (hunk 1 above)
Comment (jwnasambu) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/OpenConceptLabSteps.java:120]: @sherrif10 Kindly Am not understanding how the TokenURl passes EDITTED_TOKEN_URL parameter kindly do you mind shading some light please!

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/OpenConceptLabSteps.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import org.openmrs.contrib.qaframework.page.OpenConceptLabPage;
+import org.openmrs.contrib.qaframework.page.OpenConceptLabSuccessPage;
+import org.openmrs.contrib.qaframework.page.SubscriptionPage;
+ 
+public class OpenConceptLabSteps extends Steps {
+	
+    private OpenConceptLabPage openconceptlabpage;
+    private ConfigureMetadataPage configureMetadataPage;
+    private static final String RELEASE_DICTIONARY_URL = "https://api.staging.openconceptlab.org/users/openmrs/collections/TD-ecly/65725685/";
+    private static final String TOKEN_URL = "bd022mad6d3df24f5c42ewewa94b53a23edf6eee7r";
```
Comment (sherrif10) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/OpenConceptLabSteps.java:33]: Add another line 34
`private static final String EDITTED_TOKEN_URL = "bd022mad6d3df24f5c42ewewa94b53a23edf6fff8r";`

then use EDITTEN_TOKEN_URL on line 120.

Comment (jwnasambu) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/OpenConceptLabSteps.java]: @sherrif10 Thanks!

**Review by sherrif10 — APPROVED** [review]
Thanks @jwnasambu

**Comment by kdaud** [comment]
@jwnasambu could you make a follow up commit that adds the test status budge?

**Comment by sherrif10** [comment]
@kdaud  Pushed it here https://github.com/openmrs/openmrs-contrib-qaframework/pull/393

**Pull request merged**